e2e/docker-image/docker-image.test.ts

```diff
@@ -0,0 +1,32 @@
+import { createTenv } from '@e2e/tenv';
+
+const { container } = createTenv(__dirname);
+
+beforeAll(async () => {
+  const isCI = !!process.env.CI;
+  const image = isCI ? 'ghcr.io/ardatan/graphql-mesh:test' : 'graphql-mesh';
+
+  container({
+    name: 'graphql-mesh-test',
+    image,
+    containerPort: 4000,
+    hostPort: 4000,
+    healthcheck: [
+      'CMD-SHELL',
+      'echo "this healthcheck is weirdly stuck on checking when http://0.0.0.0:4000 is ready, so this is to bypass it"',
+    ],
+    pipeLogs: true,
+    skipImagePulling: true,
+  });
+
+  // wait for container to be ready
+  await new Promise(res => setTimeout(res, 8000));
+});
+
+it('should pass health checks', async () => {
+  const healthcheckResponse = await fetch(`http://0.0.0.0:4000/healthcheck`);
+  expect(healthcheckResponse.status).toBe(200);
+
+  const readinessResponse = await fetch(`http://0.0.0.0:4000/readiness`);
+  expect(readinessResponse.status).toBe(200);
```

Comment:
No hardcoded ports. This makes tests serial - we want to have them concurrent. 

Instead just get the port from the container
```ts
const { port }  = await container(...);
```

Reply:
the issue is that whenever I `await`, it gets stuck forever